spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -778,15 +816,30 @@ def __regroup_per_tracts(self, vol_dct, tracts):
         return np.sum(res_mask) * 100.0 / np.sum(res_tot)
 
     def _measure_totLesion_distribution(self, im_lesion, atlas_data, im_vert, p_lst):
-
+        """the region is the reference (the percentage of the region affected by the lesion)"""
         sheet_name = 'ROI_occupied_by_lesion'
         total_row = f'total % (all {self.row_name})'
         rows_with_total = {
             **self.rows,
             # numpy array index equivalent to [:, :, :]
             total_row: (slice(None), slice(None), slice(None)),
         }
-        self.distrib_matrix_dct[sheet_name] = pd.DataFrame.from_dict({'row': [str(r) for r in rows_with_total]})
+
+        # Create the initial DataFrame with row column
+        df_data = {'row': [str(r) for r in rows_with_total]}
+
+        # Add vertebral level column when using per-slice analysis
+        if self.row_name == "slice":
+            vert_levels = []
+            for row_key in rows_with_total.keys():
+                if row_key == total_row:
+                    vert_levels.append('Total')
+                else:
+                    vert_level = self._get_vertebral_level_for_slice(im_vert, row_key)
+                    vert_levels.append(vert_level)
+            df_data['vert_level'] = vert_levels
+
+        self.distrib_matrix_dct[sheet_name] = pd.DataFrame.from_dict(df_data)
```

Comment:
I think we can simplify this a lot by using pandas' `apply` functionality:

```python
        df_data = {'row': [str(r) for r in rows_with_total]}
        if self.row_name == "slice":
            df_data['vert_level'] = df['row'].apply(self._get_vertebral_level_for_slice, args=im_vert)
        self.distrib_matrix_dct[sheet_name] = pd.DataFrame.from_dict(df_data)
```

I think this might require that we change the order of the arguments of the `

The benefit of using `apply` is that it